### ISSUE TITLE
Fix build process after upgrade to core18

### ISSUE THREAD

**Issue by vpetersson** [WoTTsecurity/wott-agent-snap#4]
The build process is currently broken. After upgrading to [Core 18](https://github.com/WoTTsecurity/wott-agent-snap/blob/master/snapcraft.yaml#L6) as the base, some issues went away, and others were introduced. As it stands right now, we cannot release an update to the Snap because of this, so it needs to be fixed.

In summary, here's the backstory:

 * Core 16 used to work until we introduced `nmap`. After spending a significant amount trying to troubleshoot this, I decided to upgrade to Core 18
 * Core 18 solved some issues, but introduced other new issues that needs to be resolved.

**Comment by a-martynovich** [comment]
@vpetersson which issues?

**Comment by vpetersson** [comment]
Give it a spin. It will essentially fail right now. [snapcraft.yaml](https://github.com/WoTTsecurity/wott-agent-snap/blob/master/snapcraft.yaml) is a bit of a science by itself, but it's important that we get it to build and work properly for strategic reasons.

**Comment by vpetersson** [comment]
Also, @a-martynovich, here's the current build log:
https://launchpadlibrarian.net/428097193/buildlog_snap_ubuntu_xenial_armhf_wott-agent_BUILDING.txt.gz

**Comment by a-martynovich** [comment]
@vpetersson 
I fixed the build on Core 16. Some changes need to be made to agent, one of the sensitive ones is version format change. Snap environment doesn't like our format, specifically the `~HASH` at the end. It may have something to do with pip (it says "dependency wott_agent was not satisified" even thouh it sees wott_agent installed).
So for Snap builds we could just use `VERSION.BUILD_NUMBER` without hash. Where do we get build number if we're building on Launchpad - I have no idea. 
I made a static hardcoded version 0.1.5.0 and it was successfully built.

Also snap version is currently specified manually in snapcraft.yaml. I'm not sure we can automate this for Launchpad.
